cmd/infracost/register.go

```diff
@@ -143,88 +144,73 @@ func registerCmd(ctx *config.RunContext) *cobra.Command {
 }
 
 func promptForName() (string, error) {
-	p := promptui.Prompt{
-		Label: "Name",
-		Validate: func(input string) error {
-			input = strings.TrimSpace(input)
-			if input == "" {
-				return errors.New("Please enter a name")
-			}
-			return nil
-		},
-	}
-	name, err := p.Run()
-	name = strings.TrimSpace(name)
+	name, err := stringPrompt("Name", func(input string) error {
+		input = strings.TrimSpace(input)
+		if input == "" {
+			return errors.New("Please enter a name")
+		}
+		return nil
+	})
```

Comment:
Can we extract these funcs to the ui pkg so they're inline with the interface

Reply:
by that i mean the validate func

Reply:
so:

```
func(input string) error {
		input = strings.TrimSpace(input)
		if input == "" {
			return errors.New("Please enter a name")
		}
		return nil
	}
```

Reply:
I'm not sure I follow. `ui` provides prompts to be used anywhere, but these validate functions are unique for `register` command. I would see such function having "Please enter a <any field name>" that takes an argument with the error message, and it seems redundant. It may also make sense to introduce `required` flag for the `StringPrompt`.

Reply:
my preference would be to have this as exported funcs in the ui pkg like so:

```
var (
   ValidateName ValidateFn = func(input string) error {
		input = strings.TrimSpace(input)
		if input == "" {
			return errors.New("Please enter a name")
		}
		return nil
    }
    ValidateEmail ValidateFn = func(input string) error {
		if input == "" {
			return errors.New("Please enter an email")
		}
		match, err := regexp.MatchString(".+@.+", input)
		if err != nil {
			return errors.New("Unable to validate email")
		}
		if !match {
			return errors.New("Please enter a valid email")
		}
		return nil
	}
)

type ValidateFn func(input string) error

```

this seems quite generic to me

Reply:
but if you feel more comfortable with it stuffed in register the thats ok with me

Reply:
I see, but we still need to provide a `label` for the prompts. So having for example `StringPrompt("Foobar", ui.ValidateName)` will show `Please enter a name` instead of `Please enter a foobar`. With the email validation it may make more sense introducing `EmailPrompt` function so there is no disconnect between the label and error messages.

Reply:
yeah but reading `StringPrompt("Foobar", ui.ValidateName)` you'd realise validateName isn't the valid validate function. So you'd just create another function. If you really want to make it generic - which I don't feel theres too much benefit in this case then you can just have a function that decorates the validate function. e.g:


```go

func ValidateStringNotEmpty(emptyMsg string) ValidateFn {
    return func(input string) error {
		input = strings.TrimSpace(input)
		if input == "" {
			return errors.New(emptyMsg)
		}
		return nil
    }
}
```

this would be used as:

```go
ui.StringPrompt("Foobar", ui.ValidateStringNotEmpty("foobar cannot be empty")
```

Reply:
let's just stick with the original functionality tho

Reply:
Yeah, I guess we can iterate on that when we need these simple prompts in other places 👍